src/UserSetting.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2019 Transposit Corporation. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Transposit } from ".";
+
+export class UserSetting {
+  private transposit: Transposit;
+  constructor(transposit: Transposit) {
+    this.transposit = transposit;
+  }
+
+  async get(key: string): Promise<any> {
```

Comment:
Should we make these generic? People can always pass in `any` if they want to opt-out.

If we do, we should change our other interfaces as well.

Reply:
Yeah, we should (and I had intended to, before I forgot), and we actually did for env.get(). I'd argue, though, that we should do this for env and userSetting but not for stash, because env and userSetting have defined schemas with defined types so developers can know what type to expect out of it, but stash as a freeform data store doesn't provide that.

Reply:
Sounds good!

Reply:
I added generic typing to get. I didn't know if I should add it to put; the compiler would only validate that they pass in the type that they expect, but they could do the same thing by declaring a type for the object they pass in. I didn't know if that would add much benefit.

Reply:
👍agreed, doesn't seem to add much for put()